std/algorithm/searching.d

```diff
@@ -3462,6 +3462,59 @@ unittest
 }
 
 /**
+Computes the index of the first occurrence of `range`'s minimum (respectively maximum).
+
+Params:
+    pred = The ordering predicate to use to determine the extremum (minimum or
+           maximum) element.
+    range = The input range to search.
+
+Returns:
+    The index of the first encounter of the minimum (respectively maximum) in `range`. If the
+    range is empty, -1 is returned.
+ */
+ptrdiff_t minPosIndex(alias pred = "a < b", Range)(Range range)
+    if (isInputRange!Range && !isInfinite!Range &&
+        is(typeof(binaryFun!pred(range.front, range.front))))
+{
+    if (range.empty) return -1;
+
+    ptrdiff_t min_pos = 0, cur_pos = 0;
+    Unqual!(typeof(range.front)) min = range.front;
```

Comment:
Don't do this because not all types are copyable to their unqualified type. Use a second range that trails the first one and is positioned at the extremum found so far.